frontends/web/src/utils/request.ts

```diff
@@ -112,3 +112,15 @@ export function apiPost(
     body: JSON.stringify(body)
   }).then(response => response.json()).then(handleError(endpoint));
 }
+
+export function generateQueryString<T extends Record<string, unknown>>(params: T): string {
+  const filteredAndSortedParams = Object.keys(params)
+    .filter(key => params[key] != null) // filters out null / undefined values
+    .sort() // sorts keys alphabetically
```

Comment:
(im assuming) this `sort` isnt strictly necessary (from backend POV). Just mimicking `query-string` library which sorts it alphabetically. We could remove this if we want.